pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -291,6 +291,60 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
         return measure.probs(obs_wires);
     }
 
+    /**
+     * @brief Probabilities with shot-noise for a subset of the full system.
+     *
+     * @param num_shots Number of shots.
+     * @param wires Wires will restrict probabilities to a subset
+     * of the full system.
+     *
+     * @return Floating point std::vector with probabilities.
+     */
+    auto probs(const std::vector<size_t> &wires, const size_t &num_shots)
```

Comment:
It can be updated as follows,
```suggestion
    auto probs(const std::vector<size_t> &wires, size_t num_shots)
```
Compilers will optimize unnecessary value copies of arguments for small and simple types.

Reply:
> Compilers will optimize unnecessary value copies of arguments for small and simple types.

Just a side note, but I'm always looking to learn more about C++ and am curious what you mean by this concretely, could you elaborate? :) 

I think I would rather argue that an integer is the smallest unit of data you can copy anyways. Aiming to avoid a copy by using a const reference thus brings no gain, since it incurs a copy (for the pointer) _and_ an additional indirection.